### PR TITLE
Added support for json functions

### PR DESCRIPTION
**Pull request by ritwizsinha**
Addresses #516

### CODE REVIEW

**Comment by ritwizsinha** [comment]
@JelteF I have added all the json functions present in [duckdb](https://duckdb.org/docs/data/json/json_functions#json-scalar-functions) to the sql file, is this fine or do we need to break this up. All of this is quite repeated I doubt a big pr would be a problem

File: sql/pg_duckdb--0.2.0--0.3.0.sql [hunk 1]
```diff
@@ -17,3 +17,227 @@ CREATE AGGREGATE @extschema@.approx_count_distinct(anyelement)
 
 CREATE DOMAIN pg_catalog.blob AS bytea;
 COMMENT ON DOMAIN pg_catalog.blob IS 'The DuckDB BLOB alias for BYTEA';
+
+-- json_exists
+CREATE FUNCTION @extschema@.json_exists(json VARCHAR , path VARCHAR)
```
Comment (JelteF) [inline sql/pg_duckdb--0.2.0--0.3.0.sql:22]: It seems confusing that these functions expect a `VARCHAR` type for the `json` argument. I'd expect that argument to be of the `json` or `jsonb` (or have functions for both of them).

File: src/pgduckdb_metadata_cache.cpp [hunk 2]
```diff
@@ -109,8 +109,9 @@ BuildDuckdbOnlyFunctions() {
 	 * each of the found functions is actually part of our extension before
 	 * caching its OID as a DuckDB-only function.
 	 */
-	const char *function_names[] = {"read_parquet",      "read_csv",   "iceberg_scan", "iceberg_metadata",
-	                                "iceberg_snapshots", "delta_scan", "read_json",    "approx_count_distinct"};
+	const char *function_names[] = {"read_parquet",      "read_csv",     "iceberg_scan",       "iceberg_metadata",
+	                                "iceberg_snapshots", "delta_scan",   "read_json",          "approx_count_distinct",
+	                                "json_exists",       "json_extract", "json_extract_string"};
```
Comment (JelteF) [inline src/pgduckdb_metadata_cache.cpp:114]: It seems this list is missing a bunch of the functions that you added.

File: test/regression/sql/json_functions_duckdb.sql [hunk 3]
```diff
@@ -0,0 +1,144 @@
+-- <JSON_EXISTS>
+-- Test 1: Path exists in a simple JSON object
+SELECT json_exists('{"a": {"b": 1}}', '$.a.b'); -- Expected: true
```
Comment (JelteF) [inline test/regression/sql/json_functions_duckdb.sql:3]: I think most of these tests simply test that DuckDB works correctly, not so much the pg_duckdb intgration.

I think for each function it should give us enough confidence into pg_duckdb to add one test with reasonable arguments, and then tests with the various combinations of `NULL` arguments. So while I really appreciate the effort to add extensive tests, I don't think they add much value and would in practice only increase test runtime.

**Review by JelteF — CHANGES_REQUESTED** [review]

File: test/regression/sql/json_functions_duckdb.sql (hunk 3 above)
Comment (ritwizsinha) [reply test/regression/sql/json_functions_duckdb.sql]: Reduced the tests

File: src/pgduckdb_metadata_cache.cpp (hunk 2 above)
Comment (ritwizsinha) [reply src/pgduckdb_metadata_cache.cpp]: Added

**Comment by ritwizsinha** [comment]
@JelteF structures are not supported as types in pg_duckdb so not possible to implement functions like:
```
json_transform(json, structure)	Transform json according to the specified structure.
from_json(json, structure)	Alias for json_transform.
json_transform_strict(json, structure)	Same as json_transform, but throws an error when type casting fails.
from_json_strict(json, structure)	Alias for json_transform_strict.
```

Added all the other ones

**Comment by ritwizsinha** [comment]
Curious why this fails on postgres 17

File: sql/pg_duckdb--0.2.0--0.3.0.sql [hunk 4]
```diff
@@ -17,3 +17,226 @@ CREATE AGGREGATE @extschema@.approx_count_distinct(anyelement)
...
 COMMENT ON DOMAIN pg_catalog.blob IS 'The DuckDB BLOB alias for BYTEA';
+
+-- json_exists
+CREATE FUNCTION @extschema@.json_exists("json" json , path VARCHAR)
+RETURNS boolean LANGUAGE 'plpgsql'
+SET search_path = pg_catalog, pg_temp
+AS
+$func$
+BEGIN
+    RAISE EXCEPTION 'Function `jsonb_exists(json, VARCHAR)` only works with Duckdb execution.';
```
Comment (JelteF) [inline sql/pg_duckdb--0.2.0--0.3.0.sql:28]: I just merged #531. Which introduces a more re-usable way of doing this:

```
AS 'MODULE_PATHNAME', 'duckdb_only_function'
LANGUAGE C;
```

Comment (ritwizsinha) [reply sql/pg_duckdb--0.2.0--0.3.0.sql]: Fixed

**Review by JelteF — APPROVED** [review]
I played around with this a bunch and added a few more tests and fixed a bunch of issues I found. I now think this is good enough to merge. @Y-- @mkaruza can either of you do another small review.

**Review by Y-- — APPROVED** [review]
